libs-scala/scala-utils/src/main/scala/com/daml/scalautil/concurrent/ExecutionContextOf.scala

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.scalautil.concurrent
+
+import scala.language.higherKinds
+import scala.{concurrent => sc}
+
+sealed abstract class ExecutionContextOf {
+  type T[+P] <: sc.ExecutionContext
```

Comment:
We can sensibly incorporate `ExecutionContextExecutor` in this scheme by writing

```scala
  type T[+P, +A] <: A
```

and defining aliases accordingly. It would then be a little closer to `scalaz.Tag`, albeit with subtyping, as well.  I'm not sure support for such subtypes is needed, though.